src/AutoRest.CSharp/Common/Input/Source/SourceInputModel.cs

```diff
@@ -56,6 +84,46 @@ public ModelTypeMapping CreateForModel(INamedTypeSymbol? symbol)
             return new ModelTypeMapping(_modelAttribute, _schemaMemberNameAttribute, symbol);
         }
 
+        public IMethodSymbol? FindForMethod(string name, IEnumerable<string> parameters)
+        {
+            var methods = _methodSet.Where(m => m.Name == name);
+            if (methods == null || methods.Count() == 0)
+            {
+                return null;
+            }
+            else if (methods.Count() == 1)
+            {
+                return methods.First();
+            }
+            else
+            {
+                foreach (var method in methods)
+                {
+                    if (method.Parameters.Count() - 1 != parameters.Count())
+                    {
+                        continue;
+                    }
+
+                    var existingParameters = method.Parameters.ToArray();
+                    int index = 0;
+                    foreach (var parameter in parameters)
+                    {
+                        // It would be better to compare the types.
+                        if (parameter != existingParameters[index++].Name)
```

Comment:
If only the last one of `parameters` has the different name when compared with the penultimate of `existingParameters`, the increment operator will also be executed and index will be equal to the length of the parameters, is this expected?

Reply:
Not expected. Fixed.